crates/primitives/src/transaction/signature.rs

```diff
@@ -17,41 +17,34 @@ pub struct Signature {
 }
 
 impl Signature {
-    /// Encode the `v`, `r`, `s` values without a RLP header.
-    /// Encodes the `v` value using the legacy scheme without EIP-155.
-    pub(crate) fn encode_inner_legacy(&self, out: &mut dyn reth_rlp::BufMut) {
-        (self.odd_y_parity as u8 + 27).encode(out);
-        self.r.encode(out);
-        self.s.encode(out);
-    }
-
     /// Output the length of the signature without the length of the RLP header, using the legacy
-    /// scheme without EIP-155.
-    pub(crate) fn payload_len_legacy(&self) -> usize {
-        (self.odd_y_parity as u8 + 27).length() + self.r.length() + self.s.length()
+    /// scheme with EIP-155 support depends on chain_id.
+    pub(crate) fn payload_len_legacy(&self, chain_id: Option<u64>) -> usize {
+        self.v(chain_id).length() + self.r.length() + self.s.length()
```

Comment:
@mattsse Need your final feedback. Thank you for consistent advices.

`payload_len_legacy()` is revised depending on `chain_id`.  Should I rename it as `payload_len_with_eip155_chain_id()` also?

Reply:
thank you so much!

yes `payload_len_with_eip155_chain_id ` sgtm